### PR TITLE
Encd 6230 fcc calc

### PR DESCRIPTION
**Pull request by gabdank**

### CODE REVIEW

File: src/encoded/searches/configs.py [hunk 1]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
...
+# with effectively the FunctionalCharacterizationExperiment config.
+@search_config(
+    name='FunctionalCharacterization'
+)
+def functional_characterization_data_view():
+    return {
+        'facets': {
+            "assay_slims": {
+                "title": "Assay type"
+            },
```
Comment (keenangraham) [inline src/encoded/searches/configs.py]: Single quotes.

File: src/encoded/searches/configs.py [hunk 2]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
...
+            'replicates.library.biosample.applied_modifications.MOI': {
+                'title': 'Multiplicity of infection'
+            },
+            'replicates.library.biosample.donor.organism.scientific_name': {
+                'title': 'Organism',
+                'open_on_load': True
+            },
+            "replicates.library.biosample.life_stage": {
+                "title": "Life stage"
```
Comment (keenangraham) [inline src/encoded/searches/configs.py]: Single quotes.

File: src/encoded/searches/configs.py [hunk 3]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
...
+            },
+            'control_type': {
+                'type': 'exists',
+                'title': 'Hide control experiments'
+            },
+            'award.project': {
+                'title': 'Project'
+            },
+            "award.rfa": {
+                "title": "RFA"
```
Comment (keenangraham) [inline src/encoded/searches/configs.py]: Single quotes.

File: src/encoded/searches/configs.py [hunk 4]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
...
+                "facet_fields": [
+                    "status",
+                    "audit.ERROR.category",
+                    "audit.NOT_COMPLIANT.category",
+                    "audit.WARNING.category",
+                    "audit.INTERNAL_ACTION.category"
+                ]
+            }
+        ]
+    }
```
Comment (keenangraham) [inline src/encoded/searches/configs.py]: Single quotes.

File: src/encoded/searches/configs.py [hunk 5]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
...
+
+
+
```
Comment (keenangraham) [inline src/encoded/searches/configs.py]: Remove line.

File: src/encoded/searches/configs.py [hunk 6]
```diff
@@ -89,6 +89,173 @@ def custom_columns_config():
     }
 
 
+# To allow FunctionalCharacterizationSeries columns properties to pass through for the search:
+# type=FunctionalCharacterizationExperiment&type=FunctionalCharacterizationSeries&type=TransgenicEnhancerExperiment
+# with effectively the FunctionalCharacterizationExperiment config.
```
Comment (keenangraham) [inline src/encoded/searches/configs.py:94]: We can't just use `config=FunctionalCharacterizationExperimentFacets&config=FunctionalCharacterizationExperimentFacetGroups` now right? This custom one provides something different?

File: src/encoded/types/dataset.py [hunk 7]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
         'elements_references.files',
     ]
 
+    @calculated_property(condition='related_datasets', schema={
```
Comment (keenangraham) [inline src/encoded/types/dataset.py:1257]: Can put condition and schema on newlines.

File: src/encoded/types/dataset.py [hunk 8]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+        "title": "Examined loci",
+        "type": "array",
+        "items": {
+            "type": "object",
+        },
+        "notSubmittable": True,
```
Comment (keenangraham) [inline src/encoded/types/dataset.py:1263]: Single quotes.

File: src/encoded/types/dataset.py [hunk 9]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+    })
+    def examined_loci(self, request, related_datasets=None):
+        if related_datasets is not None:
+            examined_loci = []
+            for related_dataset in related_datasets:
+                related_datasetObject = request.embed(related_dataset, '@@object?skip_calculated=true')
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: Snakecase.

File: src/encoded/types/dataset.py [hunk 10]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+                dataset_examined_loci = related_datasetObject.get('examined_loci')
+                if dataset_examined_loci:
+                    examined_loci.extend(dataset_examined_loci)
+            if examined_loci:
+                return examined_loci
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: Are the items always unique or could there be duplicates?

File: src/encoded/types/dataset.py [hunk 11]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+    @calculated_property(condition='related_datasets', schema={
+        "title": "Examined loci",
+        "type": "array",
+        "items": {
+            "type": "object",
+        },
+        "notSubmittable": True,
+    })
+    def examined_loci(self, request, related_datasets=None):
+        if related_datasets is not None:
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: Looks like `condition` is related_datasets so shouldn't need this.

File: src/encoded/types/dataset.py [hunk 12]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
 
+    @calculated_property(condition='related_datasets', schema={
+        "title": "Examined loci",
+        "type": "array",
+        "items": {
+            "type": "object",
+        },
+        "notSubmittable": True,
+    })
+    def examined_loci(self, request, related_datasets=None):
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: Would something like `return request.select_distinct_values('examined_loci', *related_datasets)` work the same way?

File: src/encoded/types/dataset.py [hunk 13]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+                '@id',
+                '@type',
+                'library',
+                'biosample',
+                'donor',
+                'treatments',
+                'applied_modifications',
+                'life_stage',
+                'disease_term_name',
+                ]
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: De-indent a little.

File: src/encoded/types/dataset.py [hunk 14]
```diff
@@ -1386,6 +1386,86 @@ class FunctionalCharacterizationSeries(Series):
...
+            )
```
Comment (keenangraham) [inline src/encoded/types/dataset.py]: Misaligned.

**Review by keenangraham — CHANGES_REQUESTED** [review]
Looks good but a couple formatting things. Also wondering if there is simpler way for `examined_loci` and if we still need custom config.

File: src/encoded/searches/configs.py (hunk 5 above)
Comment (gabdank) [reply src/encoded/searches/configs.py]: fixed

File: src/encoded/searches/configs.py (hunk 4 above)
Comment (gabdank) [reply src/encoded/searches/configs.py]: fixed all the single quotes

File: src/encoded/types/dataset.py (hunk 11 above)
Comment (gabdank) [reply src/encoded/types/dataset.py]: updated

File: src/encoded/types/dataset.py (hunk 13 above)
Comment (gabdank) [reply src/encoded/types/dataset.py]: fixed

File: src/encoded/types/dataset.py (hunk 14 above)
Comment (gabdank) [reply src/encoded/types/dataset.py]: fixed

File: src/encoded/types/dataset.py (hunk 12 above)
Comment (gabdank) [reply src/encoded/types/dataset.py]: nope it fails

Comment (keenangraham) [reply src/encoded/types/dataset.py]: Probably need a concrete field that could be deduplicated. Looks like `examined_loci` doesn't have `@id` or anything.

**Review by keenangraham — DISMISSED** [review]
A couple places where `"` -> `'` for consistency.

**Review by hitz — APPROVED** [review]
quick review after rebase.

**Pull request merged**